Java/src/main/java/com/example/aggregator/CharCounter.java

```diff
@@ -51,8 +56,42 @@ public Topology buildTopology() {
                 Materialized.with(Serdes.Integer(), Serdes.Integer())
             )
             .toStream()
-            .foreach((k, v) -> LOGGER.info("Count for {} is {}", k, v));
-        return builder.build();
+            .foreach((k, v) -> LOGGER.info("Total count of characters = {}", v));
+    }
+
+    private void addCharCounter(StreamsBuilder streamsBuilder) {
+        streamsBuilder
+            .stream(Common.CHAT_TOPIC, Consumed.with(Serdes.String(), Serdes.String()))
+            .groupByKey(Grouped.with(Serdes.String(), Serdes.String()))
+            .aggregate(
+                () -> 0, 
+                (k, msg, agg) -> agg + msg.length(),
+                Materialized.with(Serdes.String(), Serdes.Integer())
+            )
+            .toStream()
+            .foreach((k, v) -> LOGGER.info("Char count for {} = {}", k, v));
+    }
+
+    private void addWordCounter(StreamsBuilder streamsBuilder) {

```

Comment:
I don't understand: does this counts the words or the messages?

Reply:
This is indeed a message counter. I'll rename this method and add a real word counter.